.github/workflows/build-containers.yml

```diff
@@ -0,0 +1,139 @@
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          
+      - name: Check if PR-Base prebuilt
+        id: check-images
+        run: |
+          build_base_missing=$(docker manifest inspect ${{ steps.vars.outputs.container_registry }}/${{ steps.vars.outputs.cr_user }}/${{ steps.vars.outputs.platform_build_base }}:${{ steps.vars.outputs.base_ver }} > /dev/null ; echo $?)
+          build_prbase_missing=$(docker manifest inspect ${{ steps.vars.outputs.container_registry }}/${{ steps.vars.outputs.cr_user }}/${{ steps.vars.outputs.platform_build }}:${{ steps.vars.outputs.build_prbase_label }} > /dev/null ; echo $?)
+          echo build_base_missing=${build_base_missing}
+          echo build_prbase_missing=${build_prbase_missing}
+          echo ::set-output name=build_base_missing::${build_base_missing}
+          echo ::set-output name=build_prbase_missing::${build_prbase_missing}
+         
+      # build base only if missing, and pr base image is missing, otherwise don't need it.
+      - name: Build build base image
+        if: ${{ steps.check-images.outputs.build_base_missing == '1' && steps.check-images.outputs.build_prbase_missing == '1' }}
+        uses: docker/build-push-action@v2
+        with:
+          context: dockerfiles/${{ steps.vars.outputs.platform_build_base }}
+          builder: ${{ steps.buildx.outputs.name }}
+          tags: |
+            ${{ steps.vars.outputs.container_registry }}/${{ steps.vars.outputs.cr_user }}/${{ steps.vars.outputs.platform_build_base }}:${{ steps.vars.outputs.base_ver }}
+          push: true
+
+      # build base image if missing
+      - name: Build PR-Base image 
+        if: ${{ steps.check-images.outputs.build_prbase_missing == '1' }}
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./dockerfiles/platform-build/Dockerfile
+          builder: ${{ steps.buildx.outputs.name }}
+          tags: |
+            ${{ steps.vars.outputs.container_registry }}/${{ steps.vars.outputs.cr_user }}/${{ steps.vars.outputs.platform_build }}:${{ steps.vars.outputs.build_prbase_label }}
+          push: true
+          build-args: |
+            CR_USER=${{ steps.vars.outputs.cr_user }}
+            CR_REPO=${{ steps.vars.outputs.container_registry }}
+            BASE_VER=${{ steps.vars.outputs.base_ver }}
+            BUILD_USER=${{ github.repository_owner }}
+            BUILD_TAG=${{ steps.vars.outputs.build_prbase_sha }}
+            BUILD_TYPE=${{ steps.vars.outputs.build_type }}
+            USE_CPPUNIT=${{ steps.vars.outputs.use_cppunit }}
+            BUILD_THREADS=${{ steps.vars.outputs.build_threads }}
+
+      - name: Build PR image 
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./dockerfiles/platform-build-incremental-container/Dockerfile
+          builder: ${{ steps.buildx.outputs.name }}
```

Comment:
push: false is the default (see https://github.com/docker/build-push-action).

The inention of this action is to test clean builds of a docker image only at this stage,, not to consume them by running them, hence they are not pushed anywhere.

They _could_ as part of your framework be used and trigger a regression suite on k8s somewhere, but I am not sure that will be the final route, i.e. it may be better to have the PR build the image directly on Azure which can be resourced more appropriately, and pushed to Azure CR so it is 'locally' available there etc.